.github/scripts/trymerge.py

```diff
@@ -816,42 +816,47 @@ def merge_on_green(pr_num: int, repo: GitRepo, dry_run: bool = False) -> None:
     org, project = repo.gh_owner_and_name()
     start_time = time.time()
     last_exception = ''
-    while True:
+    elapsed_time = 0.0
+    while elapsed_time < 400 * 60:
```

Comment:
curious why we have such long (6+ hours) timeout. Does merging PR also run additional CI jobs?

Reply:
No idea, 6h limits sounds wrong to me (and the job would probably be killed earlier)
@zengk95 care to explain?

Reply:
I was thinking the scenario where
Person creates PR > Triggers CI > Immediately after, they write merge this > CI can potentially take 4-5 hours to run so this script spins for around 6 hours (once we make those tests mandatory)? which only gives us a 1 or 2 hour overhead. I chose 6 hours cause I think that's the default GHA timeout. 

Or the scenario where they retrigger CI by pushing another commit after like an hour or so and still want to land it when green.